detector/cve/cve202338408/cve202338408.go

```
@@ -234,7 +234,13 @@ type fileLocations struct {
 
 func versionLessEqual(lower, upper string) bool {
 	// Version format looks like this: 3.7.1p2, 3.7, 3.2.3, 2.9p2
-	return semantic.ParsePackagistVersion(lower).CompareStr(upper) <= 0
+	r, err := semantic.MustParse(lower, "Packagist").CompareStr(upper)
+
+	if err != nil {
+		panic(err)
```

Lets not reintroduce the panic. WDYT about returning bool,error?

---

Yeah I agree but was going to do that as a follow up since that's a separate concern, since as-is we're not changing the behaviour in anyway.

An alternative would be to just ignore the error here, and then the follow up PR would be for properly handling it - that'd let us avoid using the `panic` function

---

+1. let's maintain the same behavior (keep the panic in) and then followup with the fix.

---

ok, looking forward to the followup

---

Have drafted #514 branched off this - feel free to start reviewing while we wait for this to land